src/wp-includes/class-wp-theme-json.php

```diff
@@ -1000,6 +1000,14 @@ protected static function get_blocks_metadata() {
 		$registry = WP_Block_Type_Registry::get_instance();
 		$blocks   = $registry->get_all_registered();
 
+		// Unset old blocks from static variable.
```

Comment:
We haven't been clearing unregistered blocks till now. This caused inconsistent outcomes, especially when the cache was in place.

Reply:
Good find. I'm really unsure if having stale metadata stored in the `WP_Theme_JSON` class is really a problem, as it seems like most of the places that make use of this data are doing so for validation purposes when comparing against the active `WP_Theme_JSON` data.

I am curious if you still see the behavior that you noticed if the changes from #6696 are applied.

If not removing this metadata leads to a bug, it would be best to document the specifics of the bug with steps to reproduce and propose this change outside the scope of this PR.